src/main/groovy/net/neoforged/gradleutils/GradleUtilsExtension.groovy

```diff
@@ -99,4 +113,46 @@ abstract class GradleUtilsExtension {
     Action<? extends MavenArtifactRepository> getMaven() {
         return GradleUtils.getForgeMaven()
     }
+
+    void setupCentralPublishing() {
+        if (project.rootProject.name != project.name) {
+            throw new UnsupportedOperationException('The nexus publishing plugin can only be applied on the root project!')
+        }
+
+        project.plugins.apply(NexusPublishPlugin)
+        project.extensions.configure(NexusPublishExtension) { ext ->
```

Comment:
I'm not exactly thrilled at the prospect of GradleUtils applying this plugin directly. Personally, I'd like the nexus publishing plugin to be applied by the buildscript author, and then GradleUtils configuring the plugin if its present.